javaslang/src/test/java/javaslang/collection/euler/Utils.java

```diff
@@ -48,8 +49,7 @@ public static BigInteger factorial(int n) {
     public static boolean isPrime(long val) {
         return API.Match(val).of(
                 Case($(2L), true),
-                Case($(3L), true),
-                Case($(n -> n > 3), n -> !Stream.rangeClosedBy(3, Math.sqrt(n), 2).exists(d -> n % d == 0)),
+                Case($(n -> n > 2), n -> !primes().takeWhile(d -> d <= Math.sqrt(n)).exists(d -> n % d == 0)),
```

Comment:
* isn't `n` always bigger than 2 here?
* consider extracting `Math.sqrt(n)`, as we're recalculating it every time now